pkg/ucp/backend/controller/resourcegroups/trackedresourceprocess.go

```diff
@@ -45,12 +45,17 @@ type TrackedResourceProcessController struct {
 
 	// Updater is the utility struct that can perform updates on tracked resources. This can be modified for testing.
 	updater updater
+
+	// defaultDownstream is the address of the dynamic resource provider to proxy requests to.
+	defaultDownstream *url.URL
 }
 
 // NewTrackedResourceProcessController creates a new TrackedResourceProcessController controller which is used to process resources asynchronously.
-func NewTrackedResourceProcessController(opts ctrl.Options) (ctrl.Controller, error) {
-	transport := otelhttp.NewTransport(http.DefaultTransport)
-	return &TrackedResourceProcessController{ctrl.NewBaseAsyncController(opts), trackedresource.NewUpdater(opts.StorageClient, &http.Client{Transport: transport})}, nil
+func NewTrackedResourceProcessController(opts ctrl.Options, transport http.RoundTripper, defaultDownstream *url.URL) (ctrl.Controller, error) {
+	return &TrackedResourceProcessController{
```

Comment:
defaultDownstream is not used in this function, can be dropped

Reply:
👍